src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.c

```diff
@@ -28,39 +36,380 @@ const ActorInit En_Tanron3_InitVars = {
+            effectPtr->unk_10 = *sZeroVec;
+            effectPtr->unk_1C = *sZeroVec;
+            effectPtr->unk_1C.y = -2.0f;
+            effectPtr->unk_34.x = 0.1f;
+            effectPtr->unk_34.y = 0.0f;
+            effectPtr->unk_34.z = Rand_ZeroFloat(M_PI * 2);
+            effectPtr->unk_02 = Rand_ZeroFloat(100.0f);
+            effectPtr->unk_10.x = randPlusMinusPoint5Scaled(25.0f);
+            effectPtr->unk_10.z = randPlusMinusPoint5Scaled(25.0f);
+            return;
+        }
+    }
+}
+
+void EnTanron3_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTanron3* this = THIS;
+
+    this->actor.gravity = -1.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->atCollider, &this->actor, &sCylinderInit);
+    Collider_InitAndSetCylinder(globalCtx, &this->acCollider, &this->actor, &sCylinderInit);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600DA20, &D_0600DAAC, this->jointTable, this->morphTable, 10);
+    Actor_SetScale(&this->actor, 0.02f);
+    EnTanron3_SetupAct(this, globalCtx);
+    this->actor.flags &= ~1;
+    this->currentRotationAngle = Rand_ZeroFloat(500000.0f);
+    this->waterSurfaceYPos = 430.0f;
+    sGyorg = (Boss03*)this->actor.parent;
+}
+
+void EnTanron3_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    sGyorg->unk_252--;
+}
+
+void EnTanron3_SpawnBubbles(EnTanron3* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 sPrimColor = { 100, 55, 55, 255 };
+    static Color_RGBA8 sEnvColor = { 50, 10, 10, 255 };
+    s32 i;
+    Vec3f velocity;
+    Vec3f acceleration;
+
+    for (i = 0; i < 20; i++) {
+        Matrix_InsertYRotation_f(Rand_ZeroFloat(2 * M_PI), MTXMODE_NEW);
+        Matrix_RotateStateAroundXAxis(Rand_ZeroFloat(2 * M_PI));
+        Matrix_GetStateTranslationAndScaledZ(Rand_ZeroFloat(3.0f) + 2.0f, &velocity);
+        acceleration.x = velocity.x * -0.05f;
+        acceleration.y = velocity.y * -0.05f;
+        acceleration.z = velocity.z * -0.05f;
+        EffectSsDtBubble_SpawnCustomColor(globalCtx, &this->actor.world.pos, &velocity, &acceleration, &sPrimColor,
+                                          &sEnvColor, Rand_ZeroFloat(30.0f) + 70.0f, Rand_ZeroFloat(5.0f) + 15.0f, 0);
+    }
+}
+
+void EnTanron3_SetupAct(EnTanron3* this, GlobalContext* globalCtx) {
+    this->actionFunc = EnTanron3_Act;
+    Animation_MorphToLoop(&this->skelAnime, &D_0600DAAC, -10.0f);
+    this->rotationStep = 0;
+    this->rotationScale = 5;
+    this->workTimer[TIMER_PICK_DIRECTION_OR_DIE] = 50;
+    this->actor.speedXZ = 5.0f;
+    this->speedMaxStep = 0.5f;
+    this->deviationFromCurrentPos.x = randPlusMinusPoint5Scaled(500.0f);
+    this->deviationFromCurrentPos.y = randPlusMinusPoint5Scaled(100.0f);
+    this->deviationFromCurrentPos.z = randPlusMinusPoint5Scaled(500.0f);
+    Math_Vec3f_Copy(&this->currentPos, &this->actor.world.pos);
+    this->timer = Rand_ZeroFloat(100.0f);
+}
 
-extern ColliderCylinderInit D_80BB9750;
+void EnTanron3_Act(EnTanron3* this, GlobalContext* globalCtx) {
```

Comment:
General comment about this function: the code could do with spacing out a bit. Also, this may be the only living action function, but I'd have thought we could do a better name... `Swim`?

Reply:
It also has code for flopping around on land, so I guess `SwimOrFlop` could work?